tests/brevitas_examples/test_llm.py

```diff
@@ -0,0 +1,442 @@
+# Copyright (C) 2024, Advanced Micro Devices, Inc. All rights reserved.
+# SPDX-License-Identifier: BSD-3-Clause
+
+from argparse import Namespace
+from dataclasses import dataclass
+import logging
+import os
+import shutil
+
+import numpy as np
+import onnx
+import pytest
+import torch
+
+from brevitas import config
+from brevitas_examples.llm.main import main
+from brevitas_examples.llm.main import parse_args
+from tests.marker import jit_disabled_for_export
+from tests.marker import requires_pt_ge
+
+
+def ptid2pathname(string):
+    return string.replace("/", "-").replace(":", "-")
+
+
+def allclose(x, y):
+    return np.allclose(x, y, rtol=1e-03, atol=1e+01, equal_nan=False)
+
+
+def allveryclose(x, y):
+    return np.allclose(x, y, rtol=1e-04, atol=2e+02, equal_nan=False)
+
+
+def allexact(x, y):
+    return np.allclose(x, y, rtol=0.0, atol=0.0, equal_nan=False)
+
+
+def assert_layer_types(model, exp_layer_types):
+    for key, string in exp_layer_types.items():
+        matched = False
+        layer_names = []
+        for name, layer in model.named_modules():
+            layer_names += [name]
+            if name == key:
+                matched = True
+                ltype = str(type(layer))
+                assert ltype == string, f"Expected layer type: {string}, found {ltype} for key: {key}"
+                continue
+        assert matched, f"Layer key: {key} not found in {layer_names}"
+
+
+class UpdatableNamespace(Namespace):
+
+    def update(self, **kwargs):
+        self.__dict__.update(**kwargs)
+
+
+def requires_fx(args):
+    return args.act_equalization == "fx" or args.weight_equalization or args.ln_affine_merge
+
+
+@dataclass
+class ModelAndPpl:
+    name: str
+    float_ppl: float
+    supports_fx: bool
+
+
+@pytest.fixture(
+    scope="session",
+    params=[
+        ModelAndPpl(
+            name="hf-internal-testing/tiny-random-LlamaForCausalLM",
+            float_ppl=None,
+            supports_fx=True,
+        ),
+        ModelAndPpl(
+            name="hf-internal-testing/tiny-random-MistralForCausalLM",
+            float_ppl=None,
+            supports_fx=False,
+        ),
+        #ModelAndPpl(
```

Comment:
Why is this (and later) commented?

Reply:
Test case ready for when/if we expand to MoE support in the LLM entry-point.